### PR TITLE
Fix WebGL premultiplied alpha. Fix texImage2D calls with RGB images.

### PR DESCRIPTION
**Pull request by MortimerGoro**
Fix WebGL premultiplied alpha testcases.

Fix broken textures in some WebGL demos (e.g. Three.js). This was caused by WebGL::texImage2D calls with RGB formats. Alpha must be removed from the rgba8 pixel vector before submitting the data to the GPU.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16590)

### CODE REVIEW

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /webgl/conformance-1.0.3/conformance/textures/gl-teximage.html:
  │ FAIL [expected PASS] WebGL test #38: Half the pixels in channel 0 should be >= 128,128,128. found 0%
  │   → assert_true: Half the pixels in channel 0 should be >= 128,128,128. found 0% expected true got false
  │ FAIL [expected PASS] WebGL test #40: Half the pixels in channel 1 should be >= 128,128,128. found 0%
  │   → assert_true: Half the pixels in channel 1 should be >= 128,128,128. found 0% expected true got false
  │ FAIL [expected PASS] WebGL test #42: Half the pixels in channel 2 should be >= 128,128,128. found 0%
  │   → assert_true: Half the pixels in channel 2 should be >= 128,128,128. found 0% expected true got false
  │ 
  │ reportTestResultsToHarness/<@http://web-platform.test:8000/webgl/conformance-1.0.3/resources/js-test-pre.js:88:9
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  │ test@http://web-platform.test:8000/resources/testharness.js:497:9
  │ reportTestResultsToHarness@http://web-platform.test:8000/webgl/conformance-1.0.3/resources/js-test-pre.js:87:7
  │ testFailed@http://web-platform.test:8000/webgl/conformance-1.0.3/resources/js-test-pre.js:159:5
  │ assertMsg@http://web-platform.test:8000/webgl/conformance-1.0.3/resources/js-test-pre.js:477:9
  │ runTests@http://web-platform.test:8000/webgl/conformance-1.0.3/conformance/textures/gl-teximage.html:241:7
  │ countDown@http://web-platform.test:8000/webgl/conformance-1.0.3/conformance/resources/webgl-test-utils.js:2001:7
  │ imageLoaded/<@http://web-platform.test:8000/webgl/conformance-1.0.3/conformance/resources/webgl-test-utils.js:2008:7
  └ loadImageAsync/img.onload@http://web-platform.test:8000/webgl/conformance-1.0.3/conformance/resources/webgl-test-utils.js:1983:5
```

**Comment by MortimerGoro** [comment]
I have fixed WebGL premultiplied alpha failing testcases too (some of them appear after fixing the RGB issue).
All gl-textimage.html tests pass now.

**Review by emilio — APPROVED** [review]
Looks reasonable to me, r=me with that nit.

File: components/script/dom/webglrenderingcontext.rs [hunk 1]
```diff
@@ -644,6 +643,57 @@ impl WebGLRenderingContext {
...
+                      width: u32,
+                      height: u32,
+                      unpacking_alignment: u32,
+                      source_premultiplied: bool,
+                      mut pixels: Vec<u8>) -> Vec<u8> {
+        let dest_premultiply = self.texture_unpacking_settings.get().contains(PREMULTIPLY_ALPHA);
+        if !source_premultiplied && dest_premultiply {
+            pixels = self.premultiply_pixels(internal_format, data_type, pixels);
+        }
+        else if source_premultiplied && !dest_premultiply {
```
Comment (emilio) [inline components/script/dom/webglrenderingcontext.rs]: I mean, the else should be after the brace, I'm not suggesting to change the condition.

**Comment by MortimerGoro** [comment]
It' seems like a missing byte_swap, I'll check it.

**Pull request merged**